analyzer/codechecker_analyzer/analyzer.py

```diff
@@ -77,13 +75,22 @@ def __get_analyzer_version(context, analyzer_config_map):
     versions = {}
     for _, analyzer_cfg in analyzer_config_map.items():
         analyzer_bin = analyzer_cfg.analyzer_binary
-        version = [analyzer_bin, u' --version']
+        version = [analyzer_bin, ' --version']
         try:
-            output = subprocess.check_output(shlex.split(' '.join(version)),
-                                             env=check_env,
-                                             universal_newlines=True)
+            output = subprocess.check_output(
+                shlex.split(
```

Comment:
Why do we join and split rather than just passing the list?

Reply:
In this PR I did not plan to refactor multiple stuff, the changes should reflect only what is required to run on python3.

The usage of shlex.split can be done as a separate change.